hpedockerplugin/hpe_storage_api.py

```diff
@@ -1,561 +1,658 @@
+
+        vol_mount = DEFAULT_MOUNT_VOLUME
+        if ('Opts' in contents and contents['Opts'] and
+                'mount-volume' in contents['Opts']):
+            vol_mount = str(contents['Opts']['mount-volume'])
+
+        path_info = self._etcd.get_vol_path_info(volname)
+        if path_info:
+            path_name = path_info['path']
+            connection_info = path_info['connection_info']
+            mount_dir = path_info['mount_dir']
+        else:
+            msg = (_LE('Volume unmount path info not found %s'), volname)
+            LOG.error(msg)
+            raise exception.HPEPluginUMountException(reason=msg)
+
+        # Get connector info from OS Brick
+        # TODO: retrieve use_multipath and enforce_multipath from config file
+        root_helper = 'sudo'
+
+        connector_info = connector.get_connector_properties(
+            root_helper, self._my_ip, multipath=self.use_multipath,
+            enforce_multipath=self.enforce_multipath)
+
+        # Determine if we need to unmount a previously mounted volume
+        if vol_mount is DEFAULT_MOUNT_VOLUME:
+            # unmount directory
+            fileutil.umount_dir(mount_dir)
+            # remove directory
+            fileutil.remove_dir(mount_dir)
+
+        # We're deferring the execution of the disconnect_volume as it can take
+        # substantial
+        # time (over 2 minutes) to cleanup the iscsi files
+        if connection_info:
+            LOG.info(_LI('call os brick to disconnect volume'))
+            d = threads.deferToThread(self.connector.disconnect_volume,
+                                      connection_info['data'], None)
+            d.addCallbacks(self.disconnect_volume_callback,
+                           self.disconnect_volume_error_callback)
+
+        try:
+            # Call driver to terminate the connection
+            self.hpeplugin_driver.terminate_connection(vol, connector_info)
+            LOG.info(_LI('connection_info: %(connection_info)s, '
+                         'was successfully terminated'),
+                     {'connection_info': json.dumps(connection_info)})
+        except Exception as ex:
+            msg = (_LE('connection info termination failed %s'),
+                   six.text_type(ex))
+            LOG.error(msg)
+            # Not much we can do here, so just continue on with unmount
+            # We need to ensure we update etcd path_info so the stale
+            # path does not stay around
+            # raise exception.HPEPluginUMountException(reason=msg)
+
+        # TODO: Create path_info list as we can mount the volume to multiple
+        # hosts at the same time.
+        self._etcd.update_vol(volid, 'path_info', None)
+
+        LOG.info(_LI('path for volume: %(name)s, was successfully removed: '
+                     '%(path_name)s'), {'name': volname,
+                                        'path_name': path_name})
+
+        response = json.dumps({u"Err": ''})
+        return response
+

```

Comment:
We need to remove or refactor the _lock_volume and _unlock_volume functions as they don't work as expected. They are currently not used by the plugin.